Pagination/MongoTrait/FilterTypeStrategy/Strategies/TranslatedFilterStrategy.php

```diff
@@ -39,16 +40,87 @@ public function support($type)
      */
     public function generateFilter($name, $value, $documentName)
     {
-        $metaData = $this->documentManager->getClassMetadata($documentName);
-        $collectionName = $metaData->getCollection();
+        $collection = $this->documentManager->getDocumentCollection($documentName);
+        $collectionName = $collection->getName();
         $dataBase = $this->documentManager->getDocumentDatabase($documentName);
-        $criteria = '{"'.$name.'.*.value": /.*'.$value.'.*/i}';
+        $criteria = array($name.'.*.value' =>  new \MongoRegex('/.*'.$value.'.*/i'));
+        $filter = $this->parseCriteria($criteria, $collectionName, $dataBase);
+        if (!empty($filter)) {
+
+            return $filter;
+        }
+
+        return null;
+    }
+
+
+    /**
+     * @param array    $criteria
+     * @param string   $collectionName
+     * @param Database $dataBase
+     *
+     * @return array
+     */
+    protected function parseCriteria($criteria, $collectionName, $dataBase)
+    {
+        foreach ($criteria as $key => $value) {
+            if(strpos($key, "*") !== false){
+                $res = $this->generateCriteria($key, $value, $collectionName, $dataBase);
+                if (!empty($res)) {
+                    $criteria['$or'] = $res;
+                    unset($criteria[$key]);
+                }
+            } elseif(is_array($value)) {
+                $res = $this->parseCriteria($value, $collectionName, $dataBase);
+                $criteria[$key] = $res;
+            }
+        }
+
+        return $criteria;
+    }
+
+    /**
+     * @param string   $key
+     * @param string   $value
+     * @param string   $collectionName
+     * @param Database $dataBase
+     *
+     * @return array
+     */
+    protected function generateCriteria($key, $value, $collectionName, $dataBase)
+    {
+        $map = new \MongoCode(
+            "function() {
+                    for (var key in eval('this.' + preColumn)) {
+                        emit(preColumn + '.' + key + '.' + postColumn, null);
+                    }
+            }"
+        );
+        $reduce = new \MongoCode("function(k, vals) {  return null; }");
+
+        preg_match('/(.*)\.\*.(.*)/',$key, $column);
+        $preColumnn = $column[1];
+        $postColumn = $column[2];
+
         $return = $dataBase->command(array(
```

Comment:
$return should be named $commandResult or something like this to be more explicit